kani-compiler/src/session.rs

```diff
@@ -30,9 +34,36 @@ static PANIC_HOOK: LazyLock<Box<dyn Fn(&panic::PanicInfo<'_>) + Sync + Send + 's
 
             // Print the Kani message
             eprintln!("Kani unexpectedly panicked during compilation.");
-            eprintln!(
-                "If you are seeing this message, please file an issue here: {BUG_REPORT_URL}"
+            eprintln!("Please file an issue here: {BUG_REPORT_URL}");
+        }));
+        hook
+    });
+
+// Custom panic hook when executing under json error format `--error-format=json`.
+#[allow(clippy::type_complexity)]
+static JSON_PANIC_HOOK: LazyLock<Box<dyn Fn(&panic::PanicInfo<'_>) + Sync + Send + 'static>> =
+    LazyLock::new(|| {
+        let hook = panic::take_hook();
+        panic::set_hook(Box::new(|info| {
+            // Print stack trace.
+            let msg = format!(
+                "Kani unexpectedly panicked at {}",
+                info.location().map_or(String::from("Unknown Location"), |loc| loc.to_string())
```

Comment:
What does this format as? What a panic typically prints? e.g. `{path}:{line}:{column}` (I think?)

Reply:
This is an example:
```
{"message":"Kani unexpectedly panicked at compiler/rustc_borrowck/src/region_infer/mod.rs:2196:9","code":null,"level":"error: internal compiler error","spans":[],"children":[],"rendered":"error: internal compiler error: Kani unexpectedly panicked at compiler/rustc_borrowck/src/region_infer/mod.rs:2196:9\n\n"}
```

Reply:
btw, I'm changing this message to print the panic info instead, which will print the message and the location.

Reply:
So now it will be:
```
{"message":"Kani unexpectedly panicked at panicked at 'no entry found for key', compiler/rustc_borrowck/src/region_infer/mod.rs:2196:9.","code":null,"level":"error: internal compiler error","spans":[],"children":[],"rendered":"error: internal compiler error: Kani unexpectedly panicked at panicked at 'no entry found for key', compiler/rustc_borrowck/src/region_infer/mod.rs:2196:9.\n\n"}
```